client/my-sites/checkout/composite-checkout/contact-validation.js

```diff
@@ -45,18 +45,18 @@ export default function createContactValidationCallback( {
 							'There was an error validating your contact information. Please contact support.'
 						)
 					);
-					resolve( false );
-					return;
 				}
-				if ( data.messages ) {
+				if ( data && data.messages ) {
 					showErrorMessage(
 						translate(
 							'We could not validate your contact information. Please review and update all the highlighted fields.'
 						)
 					);
 				}
-				applyDomainContactValidationResults( { ...data.messages } );
-				resolve( ! ( data.success && areRequiredFieldsNotEmpty( decoratedContactDetails ) ) );
+				applyDomainContactValidationResults( { ...( data ? data.messages : {} ) } );
```

Comment:
Minor, but couldn't this just be the same without the spread operator (not sure why we used the spread operator in the first place)? Like:

```js
applyDomainContactValidationResults( data ? data.messages : {} );
```

Or even simpler, and probably more accurate since it also tests `data.messages`:

```js
applyDomainContactValidationResults( data?.messages ?? {} );
```